kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/KafkaProxyInitializer.java

```diff
@@ -124,13 +129,15 @@ public void channelActive(ChannelHandlerContext ctx) {
     @SuppressWarnings("OptionalUsedAsFieldOrParameterType")
     private void initTlsChannel(SocketChannel ch, ChannelPipeline pipeline, Optional<String> bindingAddress, int targetPort) {
         LOGGER.debug("Adding SSL/SNI handler");
-        pipeline.addLast(new SniHandler((sniHostname, promise) -> {
+        pipeline.addLast("sniResolver", new SniHandler((sniHostname, promise) -> {
             try {
-                var stage = virtualClusterBindingResolver.resolve(Endpoint.createEndpoint(bindingAddress, targetPort, tls), sniHostname);
+                Endpoint endpoint = Endpoint.createEndpoint(bindingAddress, targetPort, tls);
+                var stage = virtualClusterBindingResolver.resolve(endpoint, sniHostname);
                 // completes the netty promise when then resolution completes (success/otherwise).
                 stage.handle((binding, t) -> {
                     try {
                         if (t != null) {
+                            LOGGER.error("Exception resolving Virtual Cluster Binding for endpoint {} and sniHostname {}: {}", endpoint, sniHostname, t.getMessage());
```

Comment:
I've always thought of ERROR as representing something in our code went wrong, and using WARN for things like miss configuration (though I probably haven't been consistent in that)

Reply:
cool, will switch to warn